src/SonarScanner.MSBuild.Shim/ScannerEngineInputGenerator.cs

```diff
@@ -415,4 +419,23 @@ string KeySuffix()
             }
         }
     }
+
+    private void AddCmdLineArgs(ScannerEngineInput engineInput)
+    {
+        var properties = cmdLineArgs.GetAllProperties().ToList();
+        properties.AddRange(SensitiveFileProperties());
+
+        foreach (var prop in properties)
+        {
+            engineInput.Add(prop.Id, prop.Value);
+        }
+
+        IEnumerable<Property> SensitiveFileProperties() =>
+            analysisConfig.AnalysisSettings(false, runtime.Logger)
+                .GetAllProperties()
+                .Where(x => x.ContainsSensitiveData() && !CmdLineArgExists(x));
+
+        bool CmdLineArgExists(Property fileProperty) =>
+            properties.Any(x => x.IsKey(fileProperty.Id));
```

Comment:
As this will be moved out, the `properties` will not be accessible. `cmdLineArgs` has `TryGetProperty` that should be used